### DISCUSSION TITLE
table sections for a better matching of Pandoc tables (>2.10)

### DISCUSSION THREAD

**Discussion by massifrg** [jgm/djot#276] [Ideas]
To support table sections and Pandoc most recent table model, is the following syntax feasible?

```
| a | b | c |
|--:|--:|--:|
| 1 | 2 | 3 |
|===|===|===|
| 2 | 4 | 6 |
```

That's a table with two bodies:
- the first one has 1 header row and a data row
- the second one has 1 data row

The logic is to use equal signs instead of minuses to divide rows into bodies.

How to code pandoc's `TableHead` and `TableFoot`?

```
|===|===|===|
| a | b | c |
|===|===|===|
| 1 | 2 | 3 |
|===|===|===|
| 2 | 4 | 6 |
```

The first section is a table head because it starts with a line containing equal signs.
The next one is for a table foot (last section ending with a line of equal signs):

```
| a | b | c |
|--:|--:|--:|
| 1 | 2 | 3 |
|===|===|===|
| 2 | 4 | 6 |
|===|===|===|
```

Bars with equal signs could also be used for horizontal alignment of the following rows:

```
|  a  |  b  |  c  |
|----:|----:|----:|
|   1 |   2 |   3 |
|:====|:===:|====:|
| 2   |  4  |   6 |
```

Pandoc `RowHeadColumns` could be obtained with a double pipe `||`

```
|     || a | b |  c |
|:----||--:|--:|---:|
| x   || 1 | 2 |  3 |
| y   || 5 | 3 |  7 |
|=====||===|===|====|
| x+y || 6 | 5 | 10 |
```

**Comment by massifrg** [comment]
I realize that even adding sections to tables it is hard to match pandoc table model in a meaningful way -- I mean in a way that is both human and machine readable.

Table cells are made of Blocks in pandoc, while they are made of Inlines in djot.
Even some inline attributes could be rather disrupting for the readability of a djot pipe table.

A complex table could be better embedded in a djot document through a HTML raw block.
Its convertibility through pandoc could be assured by a filter.

**Reply by jgm** [reply]
We definitely need to figure out how to support more complex tables in djot.  My main idea so far is that we might be able to do this using "list tables", specially marked lists whose items represent rows and columns.  But this is definitely on the roadmap.
